### PR TITLE
feat: [M3-8883] - Use our search parser on Images Landing

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Uses our search parser on the Images Landing page to provide users with slightly more flexibility  🔍
- This is just a "soft-launch".  We can add user facing documentation as a follow up once we're confident in the feature and the API has had some time to improve filterability.

### Search by `region`

This will help power users using Image Service Gen2

> [!note]
> Sadly does not support complex operations (`+or` / `+and`) due to API limitations. Only available in dev right now.

screenshot

Screenshot 2024-11-08 at 1 02 38 PM

### Search by `tags` (or `tag` because it is aliased)

Adds a sustainable way to filter by tag 🎉 

> [!note]
> Searching by tag is partially broken. See `ARB-5792`

Screenshot 2024-11-08 at 1 05 38 PM

## How to test 🧪

- Test various search queries on the Images landing page 🔍 
- Verify the UI gracefully handles valid and invalid searches ✅ ❌ 

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by abailly-akamai — APPROVED** [review]
This is great and a nice improvement 🎉  Left comments that should be addressed, but are non-blocking.
- confirmed search parity with search filter and global search ✅ 
- confirmed error handling is sound ✅ 

My only concern in general with the feature (not so now, cause not only is it a parity item but a considerable improvement) is how can we communicate this feature to the user knowing that:
- every endpoint has its quirks and may or may not support a variety of filters, including filter operators
- every endpoint has its own handling of errors (sometimes we'll get a proper 400 with a message, sometimes a 500)

This is going to be an intricate UI and I wonder if we should start thinking of a util that's going to help us generate this contextual help, taking as arguments the things it can and can't do (with which we could also perhaps return our own errors)

Screenshot 2024-11-13 at 09 32 08

Screenshot 2024-11-13 at 09 28 29

File: packages/manager/src/queries/images.ts [hunk 1]
```diff
@@ -64,6 +64,7 @@ export const useImagesQuery = (
   useQuery<ResourcePage<Image>, APIError[]>({
     ...imageQueries.paginated(params, filters),
     placeholderData: keepPreviousData,
+    retry: false,
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/queries/images.ts]: Good call. Addressed in 7c3e580

File: packages/manager/src/features/Images/ImagesLanding/ImagesLanding.tsx [hunk 2]
```diff
@@ -399,10 +425,12 @@ export const ImagesLanding = () => {
...
       <TextField
         InputProps={{
-          endAdornment: imageLabelFromParam && (
+          endAdornment: query && (
             <InputAdornment position="end">
               {isFetching && <CircleProgress size="sm" />}
-
+              {searchParseError && (
+                <TooltipIcon status="error" text={searchParseError.message} />
+              )}
```
Comment (hana-akamai) [inline packages/manager/src/features/Images/ImagesLanding/ImagesLanding.tsx:433]: Error tooltip in an input seems like a new pattern, is there a reason why we're not using the textfield errorText instead?

**Review by hana-akamai — APPROVED** [review]

File: packages/manager/src/features/Images/ImagesLanding/ImagesLanding.tsx (hunk 2 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Images/ImagesLanding/ImagesLanding.tsx:433]: That's a good point. I guess I was concerned about layout shifting but I think being consistent using `errorText` would be better

**Pull request merged**